cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -404,8 +404,12 @@ class TestAzuriteGeneric : public TestGeneric {
   }
 
  protected:
-  // Azurite doesn't support moving files over containers.
+  // Azurite doesn't block writing files over directories.
+  bool allow_write_file_over_dir() const override { return true; }
+  // Azurite doesn't support moving files.
   bool allow_move_file() const override { return false; }
+  // Azurite doesn't support moving directories.
+  bool allow_move_dir() const override { return false; }
```

Comment:
Could you keep the same order as `TestGeneric` (`allow_move_dir() -> allow_move_file()`) ?

```suggestion
  // Azurite doesn't support moving directories.
  bool allow_move_dir() const override { return false; }
  // Azurite doesn't support moving files.
  bool allow_move_file() const override { return false; }
```